compiler/rustc_hir/src/hir.rs

```diff
@@ -4445,6 +4447,70 @@ pub struct Impl<'hir> {
     pub constness: Constness,
 }
 
+impl Impl<'_> {
+    pub fn is_fully_generic_for_reflection(&self) -> bool {
+        #[derive(Default)]
+        struct LifetimeFinder {
+            seen: FxHashSet<LocalDefId>,
+        }
+        impl<'v> Visitor<'v> for LifetimeFinder {
+            type Result = ControlFlow<()>;
+            fn visit_lifetime(&mut self, lifetime: &'v Lifetime) -> Self::Result {
+                match lifetime.kind {
+                    LifetimeKind::Param(def_id) => {
+                        if self.seen.insert(def_id) {
+                            ControlFlow::Continue(())
+                        } else {
+                            ControlFlow::Break(())
+                        }
+                    }
+                    LifetimeKind::ImplicitObjectLifetimeDefault
+                    | LifetimeKind::Error
+                    | LifetimeKind::Infer
```

Comment:
Does this refer to `'_` and/or "hidden behind `&[mut]`" lifetimes? I think those are analogous to unique, non-nameable-in-bounds, lifetimes, so this ought to be fine?